test2/test2/Controllers/CountryModelsController.cs

```diff
@@ -0,0 +1,153 @@
+
+        // GET: CountryModels
+        public async Task<IActionResult> Index()
+        {
+            return View(await _context.CountryModel.ToListAsync());
+        }
+
+        // GET: CountryModels/Details/5
+        public async Task<IActionResult> Details(int? id)
+        {
+            if (id == null)
+            {
+                return NotFound();
+            }
+
+            var countryModel = await _context.CountryModel
+                .FirstOrDefaultAsync(m => m.Id == id);
+            if (countryModel == null)
+            {
+                return NotFound();
+            }
+
+            return View(countryModel);
+        }
+
+        // GET: CountryModels/Create
+        public IActionResult Create()
+        {
+            return View();
+        }
+
+        // POST: CountryModels/Create
+        // To protect from overposting attacks, enable the specific properties you want to bind to.
+        // For more details, see http://go.microsoft.com/fwlink/?LinkId=317598.
+        [HttpPost]
+        [ValidateAntiForgeryToken]
+        public async Task<IActionResult> Create([Bind("Id,Name,Salary,Area,People,Sea")] CountryModel countryModel)
+        {
+            if (ModelState.IsValid)
+            {
+                _context.Add(countryModel);
+                await _context.SaveChangesAsync();
+                return RedirectToAction(nameof(Index));
+            }
+            return View(countryModel);
+        }
+
+        // GET: CountryModels/Edit/5
+        public async Task<IActionResult> Edit(int? id)
+        {
+            if (id == null)
+            {
+                return NotFound();
+            }
+
+            var countryModel = await _context.CountryModel.FindAsync(id);
+            if (countryModel == null)
+            {
+                return NotFound();
+            }
+            return View(countryModel);
+        }
+
+        // POST: CountryModels/Edit/5
+        // To protect from overposting attacks, enable the specific properties you want to bind to.
+        // For more details, see http://go.microsoft.com/fwlink/?LinkId=317598.
+        [HttpPost]
+        [ValidateAntiForgeryToken]
+        public async Task<IActionResult> Edit(int id, [Bind("Id,Name,Salary,Area,People,Sea")] CountryModel countryModel)
+        {
+            if (id != countryModel.Id)
+            {
+                return NotFound();
+            }
+
+            if (ModelState.IsValid)
+            {
+                try
+                {
+                    _context.Update(countryModel);
+                    await _context.SaveChangesAsync();
+                }
+                catch (DbUpdateConcurrencyException)
+                {
+                    if (!CountryModelExists(countryModel.Id))
+                    {
+                        return NotFound();
+                    }
+                    else
+                    {
+                        throw;
+                    }
+                }

```

Comment:
Использование try/catch считается не самой лучшей практикой, хотя есть и необходимость.
Я бы избежал этой конструкции в пользу заблаговременной проверки на наличие модели в БД. Пример:
```suggestion
                            var country = _context.CountryModel.FirstOrDefault(c => c.Id == id);
            if (country == null)
            {
                return NotFound();
            }

            _context.Update(country);
            await _context.SaveChangesAsync();
```
Избежать всех ошибок БД не выйдет, но по поведению этот код не отличается от предложенного тобой, в случае любой ошибки кроме отсутствия модели - просто выбрасываем это исключение (в моем варианте это будет сделано изнутри метода `SaveChangesAsync` и не отловится в контроллере)